### PR TITLE
[Merged by Bors] - activation: don't ignore coinbase passed in StartSmeshing

### LINKED ISSUE #2711: Smesher rewards address not getting updated
## Repro
1. Set a rewards address 1 via the smesher api.
2. Setup post via the api and provide rewards address 2.
3. Query the node for its current reward address.

Result: address 1 is returned.
Expected: address 2 to be set as smesher rewards address (coinbase) and to be returned by the smesher api.

Reproduced on branch [healing-state-refactor](https://github.com/spacemeshos/go-spacemesh/tree/healing-state-refactor)

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

closes: https://github.com/spacemeshos/go-spacemesh/issues/2711

## TODO

- [ ] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Review by antonlerner — APPROVED** [review]

File: activation/activation.go [hunk 1]
```diff
@@ -224,6 +224,7 @@ func (b *Builder) StartSmeshing(ctx context.Context, coinbase types.Address, opt
 		b.mtx.Unlock()
 		return errors.New("already started")
 	}
+	b.coinbaseAccount = coinbase
```
Comment (antonlerner) [inline activation/activation.go:236]: you can add a UT for this :)

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**